### PR TITLE
Create repo-search sample (only targeting Android View's for now)

### PR DESCRIPTION
**Pull request by veyndan**
This is mostly a rip from [Multiplatform Paging's Repo Search sample](https://github.com/cashapp/multiplatform-paging/tree/main/samples/repo-search) (except for the search bar, which is yet to be implemented).

The sample app's purpose is to showcase (and help debug) the usage of Treehouse with Multiplatform Paging. There are currently a few bugs with the integration between the two, and a minimal example to test out ideas has been super valuable to me.

This currently only targets Android View's, but with time, other targets should be added.

Here's a demo of what it looks like. The random flashing and jumping currently exists on `trunk` when integrating Multiplatform Paging with Treehouse.

[device-2023-04-18-152354.webm](https://user-images.githubusercontent.com/6900601/232791130-942150e8-ddb7-4e71-860b-f69b49a815cf.webm)

### CODE REVIEW

File: samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt
Comment (swankjesse) [inline samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt:34]: Jesse thinks about a useful first thing to do with multiplatform OkHttp

**Review by swankjesse — APPROVED** [review]

File: samples/repo-search/schema/compose/build.gradle
Comment (JakeWharton) [inline samples/repo-search/schema/compose/build.gradle]:
```suggestion
  jvm()
```

File: samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt
Comment (JakeWharton) [inline samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt:1]: This whole file can be deleted and switched to the recently-landed impl, right?

File: samples/repo-search/presenter/build.gradle
Comment (JakeWharton) [inline samples/repo-search/presenter/build.gradle]:
```suggestion
  jvm()
```

File: samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt
Comment (JakeWharton) [inline samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt:38]: You're not actually using cancelation here. Needs something like `continuation.invokeOnCancelation { call.cancel() }` or whatever the API is...

File: samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt
Comment (JakeWharton) [inline samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt]: You need to check if `response.isSuccess` and otherwise resume with some kind of HTTP exception for non-2xx codes before reading the body. Otherwise you could be reading the body of a 404 or 500 or something.

**Review by JakeWharton — APPROVED** [review]
Needs a new CI job in `build.yaml` to match how the other samples are built. And we'll have to add it as a required job.

File: samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt
Comment (veyndan) [reply samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt:38]: Tbh I'd copy-pasted this from the [Emoji Search sample](https://github.com/cashapp/redwood/blob/trunk/samples/emoji-search/android-views/src/main/java/com/example/redwood/emojisearch/android/views/RealHostApi.kt). Whatever fixes are done here, they should probably also be done in Emoji Search. Do you think it's worth creating a separate HTTP module within the samples directory in order to avoid duplication? Usually I'd think it best, but as it's  a sample directory, I don't want to overcomplicate things and detract from the stuff that matters (i.e., the Treehouse stuff).

Comment (JakeWharton) [reply samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt:38]: Haha 😬 

Uhhh I'm not sure what we should do to be honest. I definitely don't want to go overkill on de-duplication because of the sample nature as you said.

I guess we can clean both of them up in a follow-up and then I can try and resume work on `fetch` in Zipline since that's the final solution. I have it started here: https://github.com/cashapp/zipline/compare/trunk...jw/fetch/2022-12-24.

File: samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt
Comment (veyndan) [reply samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt:1]: Forgot about this 😬 

When trying to depend on `projects.redwoodTreehouseLazylayoutPaging` I get the following error:

```
> Task :samples:repo-search:presenter-treehouse:compileDevelopmentExecutableKotlinJs FAILED
e: Module "app.cash.redwood:presenter" has a reference to symbol app.cash.redwood.treehouse.lazylayout.paging/items$composable|-6858319201313537441[0]. Neither the module itself nor its dependencies contain such declaration.

This could happen if the required dependency is missing in the project. Or if there is a dependency of "app.cash.redwood:presenter" that has a different version in the project than the version that "app.cash.redwood:presenter" was initially compiled with. Please check that the project configuration is correct and has consistent versions of all required dependencies.

…
```

Any ideas? A similar error has been preventing us from using it internally, so I'm assuming it's an issue with `projects.redwoodTreehouseLazylayoutPaging` itself.

Comment (JakeWharton) [reply samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt:1]: I'll try to look tonight so as to unblock you, but if not tomorrow first thing.

Comment (veyndan) [reply samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt:1]: @JakeWharton Thanks for offering to look into it. Did you get any time to check it out?

File: samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/AndroidRepoSearchWidgetFactory.kt
Comment (JakeWharton) [inline samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/AndroidRepoSearchWidgetFactory.kt]: Looks unused

File: samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt
Comment (JakeWharton) [reply samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt:1]: I did that night... and then forgot to pick it back up in the morning. Doing that now.

Comment (JakeWharton) [reply samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt:1]: If you just want to drop a `// TODO` we don't need to hold this up

File: samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/AndroidRepoSearchWidgetFactory.kt
Comment (veyndan) [reply samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/AndroidRepoSearchWidgetFactory.kt]: Removed

File: samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt
Comment (veyndan) [reply samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt:1]: Just added a `TODO`!

Comment (JakeWharton) [reply samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt:1]: Pushed the fix. The Compose compiler plugin needed applied to the lazy paging module because it contained composable lambdas in its functions (whose signatures need transformed to include the hidden parameters).

Comment (veyndan) [reply samples/repo-search/presenter/src/jsMain/kotlin/com/example/redwood/reposearch/presenter/LazyPagingItems.kt:1]: Sweet — thank you!

**Pull request merged**
